pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -373,12 +375,19 @@ func (lb *ListenerBuilder) buildSidecarOutboundListeners(node *model.Proxy,
 			// If captureMode is not NONE, i.e., bindToPort is false, then
 			// we will bind to user specified IP (if any) or to the VIPs of services in
 			// this egress listener.
+			var extraBind []string
 			bind := egressListener.IstioListener.Bind
 			if bind == "" {
 				if bindToPort {
-					bind = actualLocalHostAddress
+					bind = actualLocalHosts[0]
```

Comment:
It is hard to understand what 0 indicate

Reply:
there should be 2 addresses for actualLocalhosts, one is for IPv4 127.0.0.1, another is for IPv6 ::1.  And actualLocalhosts[0] is the first localhost address defined in application's service yaml file with IPFamilies.

Reply:
Why do we need `extraBind` at all? Instead of `bind` being a list?

I am very worried that somewhere else we will assume there is one bind and mutate it.

For example:

```
bind := 127.0.0.1
extraBind := ::1
... lot of code later ..
bind = SomeOtherIp // Oops! forgot to change extrabind!
```

Reply:
Hi @howardjohn, `bind == ""` is a condition to do such specification for the `bind`, if bind is not empty, it should not be some other IP. This is reasonable.

Reply:
Hi @howardjohn,  please ignore my previous answer. I understand your concerns, in fact listenerOpt's `bind` would not be specified in other place once it's available in upper function, such as `buildSidecarOutboundListeners`. But there is an except in `buildSidecarOutboundTCPListenerOptsForPortOrUDS` when `bind` is empty like below:
```
......
if len(listenerOpts.bind) == 0 {
                svcListenAddress := listenerOpts.service.GetAddressForProxy(listenerOpts.proxy)
                svcExtraListenAddress := listenerOpts.service.GetExtraAddressesForProxy(listenerOpts.proxy)
                // We should never get an empty address.
                // This is a safety guard, in case some platform adapter isn't doing things
                // properly
                if len(svcListenAddress) > 0 {
                        if !strings.Contains(svcListenAddress, "/") {
                                listenerOpts.bind = svcListenAddress
                        } else {
                                // Address is a CIDR. Fall back to 0.0.0.0 and
                                // filter chain match
                                destinationCIDR = svcListenAddress
                                listenerOpts.bind = actualWildcard
                        }
                        listenerOpts.extraBind = svcExtraListenAddress
                }
        }
......
```
so I think the situation is under the control.

Reply:
can you please add a comment on why we are using bind = actualLocalHosts[0]?

Reply:
sure, I added comment for both `bind = actualLocalHosts[0]` and `bind = actualWildcards[0]` here.